src/main/java/fr/agaetis/monobot/web/controller/PictureController.java

```diff
@@ -0,0 +1,88 @@
+package fr.agaetis.monobot.web.controller;
+
+import com.flickr4java.flickr.Flickr;
+import com.flickr4java.flickr.FlickrException;
+import com.flickr4java.flickr.photos.Photo;
+import com.flickr4java.flickr.photos.PhotoList;
+import fr.agaetis.monobot.model.mongo.Picture;
+import fr.agaetis.monobot.repository.mongo.PictureRepository;
+import fr.agaetis.monobot.service.FlickRService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Optional;
+
+@RestController
+public class PictureController {
+
+    private final FlickRService flickrService;
+    private final PictureRepository pictureRepository;
+    private Flickr flickr;
+    private ArrayList<String> urls = new ArrayList<>();
+
+
+    @Autowired
+    public PictureController(FlickRService flickrService, PictureRepository pictureRepository) {
+        this.flickrService = flickrService;
+        this.pictureRepository = pictureRepository;
+    }
+
+    /**
+     * Show and save newly added photo addresses
+     */
+    @RequestMapping("/pictures")
+    public ArrayList<String> picture() throws FlickrException {
+        this.flickr = this.flickrService.getInstance();
+        String[] groups = this.flickrService.getGroups();
```

Comment:
préférer un usage des stream Java 8 -> `return this.flickrService.getGroups().stream().flatmap(PictureController::retrievePhotoFromGroup).collect(Collectors.toList())`